torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -2317,7 +2317,7 @@ void verify_params_across_processes(
     }
   }
 
-  auto metadata_dev = metadata.clone().to(params[0].device());
+  auto metadata_dev = metadata.to(params[0].device());
```

Comment:
For safety, here. Can we add copy=True to the 'to' call so it's always copied even if the tensor device matches?

Reply:
metadata is not used after this statement.

Reply:
No, it's used and has subtensors extracted on line 2321. I'm worried it might rely on the fact that metadata_dev is almost never on param device, and it's always a copy. These two operations can be fused with `  auto metadata_dev = metadata.to(params[0].device(), copy=True);` where either it is cloned or copied to device, but not both (removing a redundant copy there.